hydra-module.nix

```diff
@@ -1,45 +1,113 @@
-{ config, pkgs, lib ? pkgs.lib, ... }:
+{ config, pkgs, lib, ... }:
 
-with lib;
+# ------------------------------------------------------------------------------
+# FIXME: send this stuff upstream to nixpkgs
+# vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
 
-let
+with rec {
+  inherit (lib) mkIf mkOption;
 
+  forceWHNF = x: builtins.seq x x;
+  forceDeep = x: builtins.deepSeq x x;
+
+  strings = lib.strings // {
+    intercalate = str: list: lib.concatStrings (lib.intersperse str list);
```

Comment:
`intercalate` already exists in the standard library as `lib.concatMapStringsSep`

Reply:
Good to know!